data/campaigns/Heir_To_The_Throne/scenarios/05b_Isle_of_the_Damned.cfg

```diff
@@ -427,7 +479,48 @@
         y=7,7,8,7,7,7,8
     [/move_unit_fake]
 
-    {PLACE_IMAGE "units/transport/galleon.png~RC(magenta>red)" 27 8}
+    {PLACE_IMAGE "units/transport/galleon.png~RC(magenta>red)~FL()" 27 8}
+
+    [unstore_unit]
+        variable=hidden_advisor
+        x,y=27,8
+    [/unstore_unit]
+
+    {CLEAR_VARIABLE hidden_advisor}
+
+    [lua]
+        code=<<
+            -- This is a bit of a hack. The issue is we can not hide the unit
+            -- with [hide_unit] because the normal [message] seems to unhide
+            -- it. So this skips a **LOT** of the work the normal [message]
+            -- does, because we know exactly what we need for here, and it
+            -- hides the unit we just got with [unstore_unit] **THEN** fires
+            -- the highlight and redraw. Since this is all a cobbled together
+            -- mess, we also need to be sure to [kill] the unit!
+
+            function wesnoth.wml_actions.message_from_hidden_unit(cfg)
+                local speaker = wesnoth.get_units(cfg)[1]
+                speaker.hidden = true
+                wesnoth.highlight_hex(speaker.x, speaker.y)
+                wesnoth.fire("redraw")
+                wesnoth.show_message_dialog({
+                    left_side=nil,
+                    title=speaker.name,
+                    message=cfg.message,
+                    portrait=speaker.portrait
+                })
+            end
+        >>
```

Comment:
Five parameters where the second is the speaker? I'm not quite sure what you mean, since `wesnoth.show_message_dialog` has no concept of a speaker, and it only takes positional parameters. The first parameter does have five possible keys, though (the last is `mirror`). Speaking of which, you should do `left_side=false` instead of `left_side=nil`. It'll have the same effect (since nil is falsy), but that's supposed to be a boolean parameter, so it's better to actually use false.